packages/builtin/array.pony

```diff
@@ -63,11 +63,11 @@ class Array[A] is Seq[A]
     """
     _ptr._offset(from_offset)._copy_to(ptr._offset(to_offset), copy_len)
 
-  fun cpointer(): Pointer[A] tag =>
+  fun cpointer(offset: USize = 0): Pointer[A] tag =>
     """
     Return the underlying C-style pointer.
     """
-    _ptr
+    _ptr._offset(offset)
```

Comment:
Maybe someone can access memory they shouldn't have access to via the combination of `cpointer` and `from_cpointer`?

In theory someone can receive any `string` or `array` and get it's `Pointer` via `cpointer` and then use that to create a new `string` or `array` pointing to a point in memory they shouldn't have access to. That is already possible without the change to taking `offset` as part of the `cpointer` function since someone could get the base pointer and then create a new `string` or `array` using `from_cpointer` with a much large `size` to access memory they shouldn't have access to. And, from what I understand, once that new `string` or `array` is created, they can then read/write the memory.